### ISSUE TITLE
DynamoDB2 global seconday indexes don't respect INCLUDE project type

### ISSUE THREAD

**Issue by qnordic** [spulec/moto#3493]
It appears moto does not respect the [ProjectionType](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Projection.html) of a global secondary index when the index is set to INCLUDE. This issue is similar to what was reported in #3055 but the fix for that issue does not appear to account for the INCLUDE type.

Here's a failing test case that demonstrates the issue. This passes when run against a real dynamo table.

```python
@mock_dynamodb2
def test_gsi_keys_only():
    table_schema = {
        "KeySchema": [{"AttributeName": "partitionKey", "KeyType": "HASH"}],
        "GlobalSecondaryIndexes": [
            {
                "IndexName": "GSI-K1",
                "KeySchema": [
                    {"AttributeName": "gsiK1PartitionKey", "KeyType": "HASH"},
                    {"AttributeName": "gsiK1SortKey", "KeyType": "RANGE"},
                ],
                "Projection": {"ProjectionType": "INCLUDE",},
            }
        ],
        "AttributeDefinitions": [
            {"AttributeName": "partitionKey", "AttributeType": "S"},
            {"AttributeName": "gsiK1PartitionKey", "AttributeType": "S"},
            {"AttributeName": "gsiK1SortKey", "AttributeType": "S"},
        ],
    }

    item = {
        "partitionKey": "pk-1",
        "gsiK1PartitionKey": "gsi-pk",
        "gsiK1SortKey": "gsi-sk",
        "someAttribute": "lore ipsum",
    }

    dynamodb = boto3.resource("dynamodb", region_name="us-east-1")
    dynamodb.create_table(
        TableName="test-table", BillingMode="PAY_PER_REQUEST", **table_schema
    )
    table = dynamodb.Table("test-table")
    table.put_item(Item=item)

    items = table.query(
        KeyConditionExpression=Key("gsiK1PartitionKey").eq("gsi-pk"),
        IndexName="GSI-K1",
    )["Items"]
    items.should.have.length_of(1)
    items[0].shouldnt.have.key("someAttribute")
```

**Comment by bblommers** [comment]
Thanks for raising this @qnordic - will mark it as an enhancement.

For whoever picks this up, the fix can hopefully be an extension to this PR: https://github.com/spulec/moto/pull/3097

**Comment by bblommers** [comment]
With #3498 merged, this is now part of moto >= 1.3.16.dev146
